cmd/lncli/cmd_pay.go

```diff
@@ -250,6 +277,16 @@ func sendPayment(ctx *cli.Context) error {
 			Amt:            ctx.Int64("amt"),
 		}
 
+		// We'll attempt to parse a payment address as well, given that
+		// if the user is uysing an AMP invoice, then they may be
+		// trying to specify that value manually.
+		payAddr, err := parsePayAddr(ctx)
```

Comment:
hmmm this is weird, not familiar with AMP yet, shouldn't the payment address be encoded in `pay_req`?

Reply:
So the idea is that we're using the invoice here just to carry details such as the destination, the final CLTV to use, hop-hints, etc. With AMP, a user can specify another pay_addr, and we use that to do a normal push payment.